content/providers/01-ai-sdk-providers/170-baseten.mdx

```diff
@@ -0,0 +1,242 @@
+
+```ts
+import { createBaseten } from '@ai-sdk/baseten';
+import { embed, embedMany } from 'ai';
+
+const baseten = createBaseten({
+  modelURL: 'https://model-xyz123.api.baseten.co/sync',
+});
+
+const embeddingModel = baseten.textEmbeddingModel();
+
+// Single embedding
+const { embedding } = await embed({
+  model: embeddingModel,
+  value: 'sunny day at the beach',
+});
+
+// Batch embeddings
+const { embeddings } = await embedMany({
+  model: embeddingModel,
+  values: [
+    'sunny day at the beach',
+    'rainy afternoon in the city',
+    'snowy mountain peak',
+  ],
+});
+```
+
+### Endpoint Support for Embeddings
+
+**Supported:**
+- `/sync` endpoints (Performance Client automatically adds `/v1/embeddings`)
+
+**Not Supported:**
+- `/sync/v1` endpoints (to avoid double `/v1` issue with Performance Client)
+- `/predict` endpoints (not compatible with Performance Client)
+
+### Performance Features
+
+The embedding implementation includes:
+- **High-performance client**: Uses `@basetenlabs/performance-client` for optimal performance
+- **Automatic batching**: Efficiently handles multiple texts in a single request
+- **Connection reuse**: Performance Client is created once and reused for all requests
+- **Built-in retries**: Automatic retry logic for failed requests
+
+## Error Handling
+
+The Baseten provider includes built-in error handling for common API errors:
+
+```ts
+import { baseten } from '@ai-sdk/baseten';
+import { generateText } from 'ai';
+
+try {
+  const { text } = await generateText({
+    model: baseten('deepseek-ai/DeepSeek-V3-0324'),
+    prompt: 'Hello, world!',
+  });
+} catch (error) {
+  console.error('Baseten API error:', error.message);
+}
+```
+
+### Common Error Scenarios
+
+```ts
+// Embeddings require a modelURL
+try {
+  baseten.textEmbeddingModel();
+} catch (error) {
+  // Error: "No model URL provided for embeddings. Please set modelURL option for embeddings."
+}
+
+// /predict endpoints are not supported for chat models
+try {
+  const baseten = createBaseten({
+    modelURL:
+      'https://model-xyz123.api.baseten.co/environments/production/predict',
+  });
+  baseten(); // This will throw an error
+} catch (error) {
+  // Error: "Not supported. You must use a /sync/v1 endpoint for chat models."
+}
+
+// /sync/v1 endpoints are not supported for embeddings (to avoid double /v1)
+try {
+  const baseten = createBaseten({
+    modelURL:
+      'https://model-xyz123.api.baseten.co/environments/production/sync/v1',
+  });
+  baseten.textEmbeddingModel(); // This will throw an error
+} catch (error) {
+  // Error: "Not supported. You must use a /sync endpoint for embeddings."
+}
+
+// /predict endpoints are not supported for embeddings
+try {
+  const baseten = createBaseten({
+    modelURL:
+      'https://model-xyz123.api.baseten.co/environments/production/predict',
```

Comment:
In the docs, let's replace xyz123 with {MODEL_ID} just so it's clearer that it needs to be replaced